src/yul.rs

```diff
@@ -351,26 +356,34 @@ impl fmt::Display for Statement {
 
 impl fmt::Display for Block {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{{")?;
-        for (_, statement) in self.statements.iter().enumerate() {
-            write!(f, " {}", statement)?;
+        let statements = self
+            .statements
+            .iter()
+            .map(|s| format!("{}", s))
+            .collect::<Vec<_>>()
+            .join("\n");
+        if statements.is_empty() {
+            write!(f, "{{ }}")
+        } else if statements.len() >= 20 || statements.contains('\n') {
```

Comment:
wat

Reply:
Put the whole block on a single line unless the formatted inner part is longer than 20 characters or contains a newline.